src/client/index.ts

```diff
@@ -265,13 +269,20 @@ export class Client {
 	}
 
 	public async disconnectWallet() {
+		let nextUI = 'start'
+		this.attestationDisconnected = true
 		let wp = await this.getWalletProvider()
-		wp.deleteConnections()
+		if (wp.getConnectedWalletData().length > 0) {
+			wp.deleteConnections()
+			nextUI = 'wallet'
+			this.attestationDisconnected = false
+		}
+
 		this.tokenStore.clearCachedTokens()
 		this.eventSender('connected-wallet', null)
 		this.eventSender('disconnected-wallet', null)
 		if (this.ui) {
-			this.ui.updateUI('wallet', { viewName: 'wallet' })
+			this.ui.updateUI(nextUI as ViewType, { viewName: nextUI })
```

Comment:
Thanks for this change.

Reply:
From testing this as you'd mentioned there is an endless loop, where the token page cannot be re-loaded which we'll need to address for this change. 

<img width="377" alt="Screen Shot 2023-03-03 at 11 53 52 am" src="https://user-images.githubusercontent.com/6808817/222604477-06b2dbc0-0bad-4b87-aea3-1b29acf62272.png">

Reply:
Could we look at applying a rule that does something like this:

- If Auto Load is on. 
- The first time they navigate to the Main view we auto load tokens.
- If they have disconnected from the wallet to start over, we do not try autoloading the tokens this time around.

This I feel is the best UX and also should remove this issue.

cc: @micwallace

Reply:
Hi guys, please ignore the bug above. This is a general but to be fixed outside of this task. 

Where when testing with autoLoadTokens false, this is the same behaviour.

cc: @snowwhitedev, @micwallace

Reply:
I've created a review PR towards this suggestion basing the changes on top of what you've done to this point in the PR.

https://github.com/TokenScript/token-negotiator/commit/997c60b7acf9cfa4ce101709c6e775776b048460

Reply:
https://github.com/TokenScript/token-negotiator/compare/for-review-only/disconnect-wallet-flow-fixes

Reply:
@nicktaras agreed. Ideally any change to the UI should be done by the UI view class. i.e. the disconnect button click handler should determine if the view needs to go back to the wallet. The only exception to this is in the disconnect-wallet event in Web3WalletProvider.